### PR TITLE
Add feature IO(e.g. File) as first argument

### PR DESCRIPTION
**Pull request by ASnow**

### CODE REVIEW

**Comment by jamesmartin** [comment]
@ASnow thanks for the PR. Could you give me some details about why you made this change?

Do you want to allow both filenames *and* IO objects? 

If you *only* want to allow IO objects, without doing any filename resolution, then you could do this in your rails app:

```ruby

#./lib/noop_asset_finder.rb
class NoopAssetFinder

  # custom asset finder that simply returns IO objects, rather than attempting to look up their path
  def self.find_asset(filename)
    if filename.is_a?(IO)
      filename
    end
  end
end

#./config/initializers/inline_svg.rb
InlineSvg.configure do |config|
  config.asset_finder = NoopAssetFinder
end

```

**Comment by ASnow** [comment]
I have admin's created files that and placed in public/system/... It don't assets_path but I want use them in templates. So, I decided to create multipurpose interface for using any svg content.

**Comment by jamesmartin** [comment]
> I use both filenames and IO

@ASnow OK. Makes sense to me. I'm happy to merge this. Could you please add a test for your change?

**Comment by ASnow** [comment]
ok, will write soon

**Comment by ASnow** [comment]
@jamesmartin, ready

File: lib/inline_svg/io_resource.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+module InlineSvg
+  module IOResource
+    def self.=== object
```
Comment (jamesmartin) [inline lib/inline_svg/io_resource.rb]: I like this pattern. 👍

**Comment by jamesmartin** [comment]
@ASnow thanks for the tests and refactoring. This looks really good to me. I'll release a new version of the gem including your changes later today.

**Comment by jamesmartin** [comment]
Released in [v0.6.3](https://github.com/jamesmartin/inline_svg/blob/4dfad3e85ce33dd6960b4022f735ed2084520078/CHANGELOG.md#063---2016-04-19).

**Pull request merged**